index.js

```diff
@@ -763,40 +687,12 @@ TChannelServerOp.prototype.sendResponse = function sendResponse(err, res1, res2)
     }
     self.responseSent = true;
     // TODO: observability hook for handler errors
-    var resFrame = self.buildResponseFrame(err, res1, res2);
-    self.sendFrame(resFrame);
+    self.callback(err, res1, res2);
 };
 
-TChannelServerOp.prototype.buildResponseFrame = function buildResponseFrame(err, res1, res2) {
-    var self = this;
-    var id = self.reqFrame.header.id;
-    var arg1 = self.reqFrame.arg1;
-    var resFrame = new v1.Frame();
-    resFrame.header.id = id;
-    resFrame.header.seq = 0;
-    if (err) {
-        // TODO should the error response contain a head ?
-        // Is there any value in sending meta data along with
-        // the error.
-        resFrame.set(isError(err) ? err.message : err, null, null);
-        resFrame.header.type = v1.Types.resError;
-    } else {
-        resFrame.set(arg1, res1, res2);
-        resFrame.header.type = v1.Types.resCompleteMessage;
-    }
-    return resFrame;
-};
-
-function isError(obj) {
-    return typeof obj === 'object' && (
-        Object.prototype.toString.call(obj) === '[object Error]' ||
-        obj instanceof Error);
-}
-
-function TChannelClientOp(options, frame, start, callback) {
+function TChannelClientOp(options, start, callback) {
     var self = this;
     self.options = options;
-    self.frame = frame;
```

Comment:
To force an encapsulation barrier, otherwise you'd have to write version specific code either within ops themselves or their users.  This forces us to create options conventions for all semantic pieces of all supported versions (essential the semantic union of all supported versions within the library).  Currently you see this as passing id, arg1, arg2, arg3, and timeout.

I'd love to see the options become ephemeral and fully decomposed onto properties on the ops themselves rather than carrying around op.options all the time.